### DISCUSSION TITLE
[Question][Hw2] About checking and updating Coverage

### DISCUSSION THREAD

**Discussion by SUNpluszero** [prosyslab-classroom/cs524-program-analysis#294] [Q&A] [answered]
Name: 전선영

Q1. While doing HW2, I failed to download the _coverage folder to local using scp command. Would you recommend more detailed information about scp or other method to check the coverage report?

Q2. Also about coverage, I manually changed the dune file at test directory to include my custom examples to enhance the coverage. Do I supposed to change the build file? I am not really familiar with the dune syntax, Can I get some recommendation about that?

**Comment by goodtaeeun** [comment, marked as answer]
A1. I believe the easiest way is to download the `_coverage` directory to your local machine. Can explain how you tried to download the `_coverage` directory? For example, "I ran `scp -r ...` in my local machine ...".

A2. Please refer to the existing test cases. Modifying `test/dune` will be sufficient. However, make sure that you have prepared all the necessary files, such as `*.input` `*.expected`, etc.

P.S. By the way, I've updated the relevant document a little bit. Check if it helps.

**Comment by m-spitfire** [comment]
What I did for 1. question was `python3 -m http.server --directory _coverage/ 80`, and `sudo ufw allow 80`, and access `<server_ip>/index.html` from my browser. I think it's more convenient from downloading coverage directory.

**Comment by goodtaeeun** [comment]
@m-spitfire Good suggestion. I will add that info to the document.

**Comment by SUNpluszero** [comment]
Thank you all for the comment! It helped me a lot!
